### DISCUSSION TITLE
Cannot connect to a broker

### DISCUSSION THREAD

**Discussion by fhainz** [adam-fowler/mqtt-nio#71] [Q&A] [answered]
Hello!

I'm trying to use this library for my project but I can't connect to a broker. This is a test app with Xcode 13

```
import SwiftUI
import MQTTNIO

@main
struct mqtttestApp: App {
    let persistenceController = PersistenceController.shared
    
    init() {
    
        let client = MQTTClient(
            host: "test.mosquitto.org",
            port: 1883,
            identifier: "My Client",
            eventLoopGroupProvider: .createNew
        )
        
        client.connect().whenComplete { result in
            print(result)
        }
        
    }
    
    var body: some Scene {
        WindowGroup {
            ContentView()
                .environment(\.managedObjectContext, persistenceController.container.viewContext)
        }
    }
}
```

I got this errors

> 2021-09-24 09:56:47.959855+0200 mqtttest[24501:2152846] [connection] nw_endpoint_handler_set_adaptive_read_handler [C1.1 5.196.95.208:1883 ready socket-flow (satisfied (Path is satisfied), viable, interface: en0, ipv4, dns)] unregister notification for read_timeout failed
2021-09-24 09:56:47.959956+0200 mqtttest[24501:2152846] [connection] nw_endpoint_handler_set_adaptive_write_handler [C1.1 5.196.95.208:1883 ready socket-flow (satisfied (Path is satisfied), viable, interface: en0, ipv4, dns)] unregister notification for write_timeout failed
success(false)

Can anybody help me?

Thank you!

**Comment by adam-fowler** [comment, marked as answer]
The output you are seeing is from Network.framework, it doesn't mean you haven't connected. But the issue you have is you've created your client as a local variable of the function. As soon as you leave the function it is deleted.

**Reply by fhainz** [reply]
The error
> success(false)

is from 
```
self.mqtt.client.connect().whenComplete { result in
    print(result)
}
```

I have created a new test project without ui. Just AppDelegate and a mqtt class.

```
import Cocoa
import MQTTNIO

@main
class AppDelegate: NSObject, NSApplicationDelegate {

    var mqtt: MQTT!

    func applicationDidFinishLaunching(_ aNotification: Notification) {
      
        self.mqtt = MQTT()
        
        self.mqtt.client.connect().whenComplete { result in
            print(result)
        }
        
    }

    func applicationWillTerminate(_ aNotification: Notification) {
        // Insert code here to tear down your application
    }

    func applicationSupportsSecureRestorableState(_ app: NSApplication) -> Bool {
        return true
    }


}
```

```
import Foundation
import MQTTNIO

class MQTT {
    
    var client: MQTTClient
    
    init() {
        
        let hostname: String = "10.0.10.1"
        let port: Int = 1883
        
        self.client = MQTTClient(
            host: hostname,
            port: port,
            identifier: "My Client",
            eventLoopGroupProvider: .createNew
        )
        
    }
    
}
``` 

Same Error.  Can you get me a short example?

**Comment by adam-fowler** [comment]
If you are getting `.success(false)` you have connected. The false indicates the server has not stored a session for you from a previous connection. If the connection had failed you would get `.failure(some error)`.

**Reply by fhainz** [reply]
Thank you!
